### PR TITLE
 drivers/nrf24l01p: misc fixed

### PR DESCRIPTION
**Pull request by kYc0o**
While compiling tests/driver_nrf24l01p_lowlevel with clang in OS X it discovers several minor issues.

This PR fixes them.

### CODE REVIEW

File: drivers/include/nrf24l01p.h [hunk 1]
```diff
@@ -36,11 +36,11 @@ extern "C" {
...
-    spi_t spi;      /**< SPI device to initialize */
-    gpio_t ce;      /**< GPIO pin to initialize as chip enable */
-    gpio_t cs;      /**< GPIO pin to initialize as chip select */
-    gpio_t irq;     /**< GPIO pin to initialize as interrupt request */
-    int listener;   /**< Place to store an ID in */
+    spi_t spi;              /**< SPI device to initialize */
+    gpio_t ce;              /**< GPIO pin to initialize as chip enable */
+    gpio_t cs;              /**< GPIO pin to initialize as chip select */
+    gpio_t irq;             /**< GPIO pin to initialize as interrupt request */
+    unsigned int listener;  /**< Place to store an ID in */
```
Comment (aabadie) [inline drivers/include/nrf24l01p.h]: Can `unsigned` be enough ?

Comment (kYc0o) [reply drivers/include/nrf24l01p.h]: maybe, will try.

Comment (kYc0o) [reply drivers/include/nrf24l01p.h]: `unsigned` also does the trick. Changed.

**Comment by kYc0o** [comment]
@aabadie I addressed your comment. Can I squash?

**Comment by kYc0o** [comment]
Ping!

**Comment by kYc0o** [comment]
ping6!

**Comment by PeterKietzmann** [comment]
Please squash. One, max two commits should be enough

**Comment by kYc0o** [comment]
Squashed.

**Pull request merged**